compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```
@@ -2266,110 +2300,152 @@ static void setLoweringConfigForComputeOps(func::FuncOp entryPointFn,
+  // sizes.
+  //
+  // Here we use the assumption in FormDispatchRegions that all ops in a
+  // dispatch have identity mapping between their parallel dimensions. So we
+  // don't need to handle the permutation on dimensions between ops except for
+  // the pack op.
+  llvm::SmallDenseMap<Operation *, SmallVector<int64_t>> reductionTileSizeMap;
+  distTileSizes.resize(maxLoopNums);
+  parallelVecTileSizes.resize(maxLoopNums);
+  bool hasSeenPackOp = false;
   for (auto op : computeOps) {
-    // The lowering config is already set on rootOperation, so we skip it.
+    assert(!hasSeenPackOp && "Pack op must be the last op");
+    // Tile sizes have been initialized from rootOperation, so we skip it.
     if (op == rootOperation)
       continue;
 
-    int numLoops = cast<TilingInterface>(op).getLoopIteratorTypes().size();
-    SmallVector<int64_t> zeros(numLoops, 0);
-    TileSizesListType tileSizesList = {distTileSizes, tileAndFuseSizes};
     TypeSwitch<Operation *>(op)
         .Case<tensor::PackOp>([&](auto packOp) {
-          ArrayRef<int64_t> innerTiles = packOp.getStaticInnerTiles();
-          ArrayRef<int64_t> dimPos = packOp.getInnerDimsPos();
-          auto outerDimsPerm = packOp.getOuterDimsPerm();
-          // Scale the outer dim tiles for pack op.
-          for (int i = 0, e = std::min<int>(tileSizesList.size(), 2); i < e;
-               ++i) {
-            auto &tileSizes = tileSizesList[i];
-            for (auto [pos, size] : llvm::zip_equal(dimPos, innerTiles)) {
-              if (tileSizes[pos] == 0 || ShapedType::isDynamic(size))
-                continue;
-              tileSizes[pos] = tileSizes[pos] / size;
-            }
-            if (!outerDimsPerm.empty())
-              applyPermutationToVector(tileSizes, outerDimsPerm);
+          if (failed(adjustTileSizesForPackOp(
+                  entryPointFn, packOp, distTileSizes, parallelVecTileSizes))) {
+            return;
```

Please do not silently drop the failure. Let's propagate it back to caller?

---

Done.